src/libmain/loggers.cc

```diff
@@ -0,0 +1,54 @@
+#include "loggers.hh"
+#include "json-logger.hh"
+#include "logging.hh"
+#include "progress-bar.hh"
+#include "util.hh"
+
+#include <atomic>
+#include <nlohmann/json.hpp>
+
+namespace nix {
+
+typedef enum {
+    logFormatRaw,
+    logFormatJSON,
+    logFormatBar,
+    logFormatBarWithLogs,
```

Comment:
What is `bar-with-logs`? Is it the same as `nix --print-build-logs / -L`?

Reply:
yes: https://github.com/tweag/nix/blob/5bc27445e686fdc1be849f4183a1ff6ada9a921c/src/nix/main.cc#L78-L82